### PR TITLE
feat: support show functions command

### LINKED ISSUE #1656: Add `SHOW FUNCTIONS` command
Add `SHOW FUNCTIONS` command to display all udf function

### PR DESCRIPTION
**Pull request by keyu813**
close #1656

### CODE REVIEW

File: hybridse/src/node/sql_node.cc [hunk 1]
```diff
@@ -70,6 +70,7 @@ static absl::flat_hash_map<CmdType, absl::string_view> CreateCmdTypeNamesMap() {
         {CmdType::kCmdShowComponents, "show components"},
         {CmdType::kCmdShowTableStatus, "show table status"},
         {CmdType::kCmdDropFunction, "drop function"},
+        {CmdType::kCmdShowFunctions, "show udf function"},
```
Comment (aceforeverd) [inline hybridse/src/node/sql_node.cc]: add a yaml case in `cases/plan/cmd.yaml` for `SHOW FUNCTIONS`. 
u can test with `ast_node_converter_test` or `planner_v2_test`

File: src/sdk/sql_cluster_router.cc [hunk 2]
```diff
@@ -1514,6 +1514,20 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::HandleSQLCmd(const h
...
+        case hybridse::node::kCmdShowFunctions: {
+            std::vector<::openmldb::common::ExternalFun> funs;
+            base::Status st = ns_ptr->ShowFunction("", &funs);
+            if (!st.OK()) {
+                *status = {::hybridse::common::StatusCode::kCmdError, "show udf function failed"};
+                return {};
+            }
+            std::vector<std::vector<std::string>> lines;
+            for (auto& fun_info : funs) {
+                std::vector<std::string> vec = {fun_info.name()};
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc]: display all fileds in `ExternalFun`

File: src/sdk/sql_cluster_router.cc [hunk 3]
```diff
@@ -1514,6 +1514,37 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::HandleSQLCmd(const h
...
+                std::vector<std::string> vec = {fun_info.name(),
+                                                openmldb::type::DataType_Name(fun_info.return_type()).substr(1),
+                                                arg_type,
+                                                is_aggregate,
+                                                fun_info.file(),
+                                                fun_info.offline_file()};
+                lines.push_back(vec);
+            }
+            return ResultSetSQL::MakeResultSet(
+                {"name", "return_type", "arg_type", "is_aggregate", "file", "offline_file"}, lines, status);
```
Comment (aceforeverd) [inline src/sdk/sql_cluster_router.cc]: first char uppercase

File: src/cmd/single_tablet_test.cc [hunk 4]
```diff
@@ -89,7 +89,26 @@ TEST_P(DBSDKTest, CreateFunction) {
                     strlength_sql,
                     int2str_sql
                 });
-    auto result = sr->ExecuteSQL("select cut2(c1), strlength(c1), int2str(c2) from t1;", &status);
+    auto result = sr->ExecuteSQL("show functions", &status);
+    if (cs->IsClusterMode()) {
+        ExpectResultSetStrEq(
+            {{"name", "return_type", "arg_type", "is_aggregate", "file", "offline_file"},
+             {"cut2", "Varchar", "Varchar", "false", "/__w/OpenMLDB/OpenMLDB/build/src/libtest_udf.so",
```
Comment (aceforeverd) [inline src/cmd/single_tablet_test.cc]: `file` filed check may omitted ? it is not constant on other environment

**Review by dl239 — APPROVED** [review]

File: src/cmd/single_tablet_test.cc (hunk 4 above)
Comment (aceforeverd) [reply src/cmd/single_tablet_test.cc]: just pass `{}` instead

Comment (dl239) [reply src/cmd/single_tablet_test.cc]: use `so_path` defined above

**Review by aceforeverd — APPROVED** [review]

File: src/cmd/single_tablet_test.cc [hunk 5]
```diff
@@ -89,7 +89,21 @@ TEST_P(DBSDKTest, CreateFunction) {
                     strlength_sql,
                     int2str_sql
                 });
-    auto result = sr->ExecuteSQL("select cut2(c1), strlength(c1), int2str(c2) from t1;", &status);
+    auto result = sr->ExecuteSQL("show functions", &status);
+    if (cs->IsClusterMode()) {
+        ExpectResultSetStrEq({{"name", "return_type", "arg_type", "is_aggregate", "file", "offline_file"},
```
Comment (aceforeverd) [inline src/cmd/single_tablet_test.cc]: test assert failed for schema name, uppercase first char

**Review by aceforeverd — CHANGES_REQUESTED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**